src/test/java/com/google/cloud/tools/gradle/appengine/core/DeployAllTaskTest.java

```diff
@@ -115,46 +118,57 @@ public void testDeployAllAction_flexible() throws AppEngineException, IOExceptio
 
     deployAllTask.deployAllAction();
 
-    assertTrue(deployConfig.getDeployables().contains(appYaml));
-    assertTrue(deployConfig.getDeployables().contains(cronYaml));
-    assertTrue(deployConfig.getDeployables().contains(dispatchYaml));
-    assertTrue(deployConfig.getDeployables().contains(dosYaml));
-    assertTrue(deployConfig.getDeployables().contains(indexYaml));
-    assertTrue(deployConfig.getDeployables().contains(queueYaml));
-    assertFalse(deployConfig.getDeployables().contains(invalidYaml));
-    verify(deploy).deploy(deployConfig);
+    assertTrue(deployCopy.getDeployables().contains(appYaml));
+    assertTrue(deployCopy.getDeployables().contains(cronYaml));
+    assertTrue(deployCopy.getDeployables().contains(dispatchYaml));
+    assertTrue(deployCopy.getDeployables().contains(dosYaml));
+    assertTrue(deployCopy.getDeployables().contains(indexYaml));
+    assertTrue(deployCopy.getDeployables().contains(queueYaml));
+    assertFalse(deployCopy.getDeployables().contains(invalidYaml));
+    verify(deploy).deploy(deployCopy);
```

Comment:
I think I might have said conflicting things earlier, but copy constructors are the desired style here,

An option while still using the copy constructor is
```
ArgumentCaptor deployCaptor = ArgumentCaptor.forClass(DeployConfiguration)
verify(deploy).deploy(deployCaptor.capture())
DeployConfiguration captured = deployCaptor.getValue()
assertTrue( //blah blah
```